app/components/trainees/filters/view.rb

```diff
@@ -32,8 +32,9 @@ def tags_for_active_filter(filter, value)
           [{ title: value, remove_link: remove_select_tag_link(filter) }]
```

Comment:
This returns the tag for an applied `subject` filter, so you probably want to add your improved title with hidden span here too.

Once you refactor `titleHtml` into a method, it will be easy to reuse.